### ISSUE TITLE
[Gen2 CF-HPG v1.2 Run] Execute preregistered tokenizer-only seed42 validation experiment

### ISSUE THREAD

**Issue by Irthn1311** [Irthn1311/FER2013_Graph#50]
## Objective

Execute exactly one preregistered **CF-HPG v1.2 Tokenizer seed-42 FER2013 train/validation run** after independent implementation review and merge of PR #49.

Primary hypothesis:

> v1.1 higher-resolution patches improved validation accuracy by +4.51 pp but still fit clean train weakly (60.54%). If the remaining bottleneck is shallow local visual encoding, adding exactly one shared nonlinear local refinement `Dense(16->96) -> GELU -> Dense(96->96)` before positional fusion should increase clean-train fit and validation performance while preserving the convolution-free/anatomy-free graph architecture.

This issue authorizes **one scientific training/validation attempt only**. It does not authorize test access, hyperparameter search, automatic retry, resume, architecture changes, or v1.3.

## Exact authorized source

Run only from exact merged `main` commit:

`032c38d49d189000b364d24af7e248eba200acf7`

Reviewed PR #49 head:

`f9e8542a5f4eb1242390d72a41f0fd718565bb92`

Implementation base / accepted v1.1 source:

`934057f60a6cb3bc7ec247e000573a40c390a426`

Registered v1.2 model source SHA-256:

`36a9a3e9e7b3ca84581a46654b592e26f71ed1cd4a156acd387417c4320986b0`

Registered model inventory:

- parameters: `420839`
- trainable variables: `66`
- Keras variables: `76`
- input `48x48x1`
- patch `4x4`, 144 fine nodes
- local tokenizer: `Dense16->96 -> GELU -> Dense96->96`
- positional projection `2->96`, additive only after raw-local refinement
- fine width 96, exactly 2 Max-Relative blocks
- hybrid spatial-8 UNION cosine-top4 graph, built once/reused twice
- exact parameter-free 2x2 arithmetic mean hierarchy to 36 coarse nodes
- coarse width 128, exactly 2 Max-Relative blocks
- same coarse hybrid graph, built once/reused twice
- global mean + max readout -> `256->128->7`

No source/model/training-config change is authorized here.

## Locked v1.1 comparison reference

Accepted Issue #47 seed42:

- clean-train accuracy `0.605385070883695`
- clean-train macro-F1 `0.5474748347135944`
- validation accuracy `0.5734187796043466`
- validation macro-F1 `0.5154247791468033`
- clean accuracy gap `3.196629127934836 pp`
- clean macro gap `3.205005556679108 pp`
- accepted label `RESOLUTION_PARTIAL_SIGNAL`

Report exact deltas:

- `delta_val_accuracy_pp = 100*(v1.2_val_acc - 0.5734187796043466)`
- `delta_val_macro_pp = 100*(v1.2_val_macro - 0.5154247791468033)`
- `delta_clean_train_accuracy_pp = 100*(v1.2_train_acc - 0.605385070883695)`
- `delta_clean_train_macro_pp = 100*(v1.2_train_macro - 0.5474748347135944)`

## Dataset boundary

Dataset signature `fer2013_train28709_val3589_test3589`.

Allowed only:
- train exactly `28709`
- validation exactly `3589`

Absolutely forbidden:
- open/read/hash/load `test.csv`
- test labels/predictions/metrics/confusion matrix
- test smoke/debug/model selection
- renaming/copying test to bypass guards

Any fail-closed path guard failure => STOP.

## Exact training configuration

Frozen from v1.1:

- seed `42`
- AdamW
- LR `3e-4`
- weight decay `5e-4`
- global clipnorm `1.0`
- batch `64`
- max epochs `100`
- linear warmup `5`
- cosine decay to `1e-6`
- CE from logits, label smoothing `0.05`
- earliest strict max `val_accuracy` checkpoint
- early stopping `val_loss`, patience `15`, min_delta `0`
- same deterministic train-only augmentation as v1.1
- validation and clean-train augmentation OFF

No MixUp/CutMix/SAM/TTA/class weights/oversampling/alternate optimizer/alternate seed.

## Exact authorized command

From exact checkout `032c38d49d189000b364d24af7e248eba200acf7`:

```bash
python -m research.candidates.tf_cf_hpg_v1_2_tokenizer.train_validation_only \
  --train-csv /kaggle/input/datasets/doduyquynii/fer13-split/fer13-split/train.csv \
  --val-csv /kaggle/input/datasets/doduyquynii/fer13-split/fer13-split/val.csv \
  --output-root /kaggle/working/tf_cf_hpg_v1_2_tokenizer_seed42
```

Do not add/change flags.

## One-attempt rule

Checkout/auth/dataset/runtime preflight stops before the training command starts do not consume the attempt.

Once the command starts, the one authorized scientific attempt is consumed. After completion, early stop, crash, OOM, CUDA failure or Kaggle censor: STOP. Do not rerun, resume or modify source/config. Any second attempt requires a new written amendment justified by technical invalidity, never by performance.

## Runtime preflight

Before launch record:
- exact HEAD and clean tracked worktree
- model SHA-256
- params / trainable-variable count / Keras-variable count
- Python / TensorFlow / Keras
- GPU model/count
- CUDA/cuDNN if available
- mixed-precision policy
- external XLA/JIT state if available

Do not externally enable mixed precision or XLA/JIT. TensorFlow-internal XLA compilation may be recorded as a runtime note.

## Registered endpoints

Primary: validation accuracy at selected earliest-strict-max-val-accuracy checkpoint.

Secondary:
- validation macro-F1/loss
- clean-train accuracy/macro-F1/loss
- accuracy/macro clean gaps
- selected epoch / epochs completed
- exact four deltas vs v1.1
- exact model identity

`acc_gap_pp = 100*(clean_train_accuracy - validation_accuracy)`

`macro_gap_pp = 100*(clean_train_macro_f1 - validation_macro_f1)`

## Frozen decision precedence

1. `CF_HPG_V1_2_STRETCH_PASS`
2. `CF_HPG_V1_2_PASS`
3. `TOKENIZER_OVERFIT_SHIFT`
4. `TOKENIZER_STRONG_SIGNAL`
5. `TOKENIZER_PARTIAL_SIGNAL`
6. `TOKENIZER_UNDERFIT_REMAINS`
7. `TOKENIZER_INCONCLUSIVE`

### STRETCH_PASS
All: val acc >= `0.7000`; val macro >= `0.6700`; both gaps <= `8 pp`.

### PASS
All: val acc >= `0.6500`; val macro >= `0.6200`; both gaps <= `8 pp`.

### TOKENIZER_OVERFIT_SHIFT
clean-train accuracy improvement vs v1.1 >= `5.0 pp` AND either clean gap > `10 pp`.

### TOKENIZER_STRONG_SIGNAL
val accuracy improvement >= `5.0 pp` AND clean-train accuracy improvement >= `5.0 pp`, unless higher-priority label applies.

### TOKENIZER_PARTIAL_SIGNAL
val accuracy improvement >= `3.0 pp` and < `5.0 pp`, unless higher-priority label applies.

### TOKENIZER_UNDERFIT_REMAINS
clean-train accuracy < `0.7000` AND validation accuracy < `0.6400`, unless higher-priority label applies.

Else `TOKENIZER_INCONCLUSIVE`.

These are single-seed practical/diagnostic labels, not statistical superiority/equivalence claims.

## Required evidence

Preserve fresh output root and full available training log. Require at minimum:
- `resolved_config.json`
- `model_summary.txt`
- `checkpoints/best_val_accuracy.keras`
- `checkpoints/best_val_accuracy.metadata.json`
- `validation_only_result.json`
- exact command / return code / runtime identity
- explicit `test_access=false`

After termination package output root once, compute ZIP SHA-256, bytes, member count, outer ZIP integrity and embedded `.keras` integrity. Do not repack after final hash.

If no valid final result exists, preserve partial artifacts as technical evidence only and do not infer a scientific label.

## Interpretation boundary

No test evaluation and no v1.3 before independent evidence review. PASS/STRETCH would justify confirmatory seeds and graph ablations; other outcomes determine the next single hypothesis only after review.

**Comment by Irthn1311** [comment]
## Independent research-lead evidence review — ACCEPTED

Reviewed the uploaded Issue #50 evidence independently against the preregistered contract, not from the execution summary alone.

### Evidence identity / integrity

- Exact checkout recorded: `032c38d49d189000b364d24af7e248eba200acf7`
- Exact model SHA-256 recorded before and after run: `36a9a3e9e7b3ca84581a46654b592e26f71ed1cd4a156acd387417c4320986b0`
- Model identity: `420839` parameters / `66` trainable variables / `76` Keras variables
- Runtime: Python 3.12.13, TensorFlow 2.20.0, Keras 3.13.2, 1x Tesla P100-PCIE-16GB, float32; external XLA/JIT field empty
- Exact authorized train/val command recorded; preflight reports clean worktree and `test_access=false`
- Downloaded evidence ZIP independently re-hashed to `e21670279e71045f0d4ad011a4dfd2a2e051a1ec27f068070b2c87202b90554a`, exactly `4,685,950` bytes
- ZIP contains exactly 7 members; independent CRC/`testzip` check passes
- Embedded checkpoint independently re-hashed to `d3a9d74bb8af8cfa3b2910ffebfe362d40f4f49d1c0e75c2edadb7ce83b08ec4`; Keras archive integrity passes and contains exactly 3 internal members
- Required artifacts are present

### Independent trajectory checks

Parsed the complete training log independently:

- exactly `91` epoch headers, from Epoch 1 through Epoch 91; no Epoch 92
- primary checkpoint maximum occurs at Epoch 90 (zero-based `89`), matching metadata (`val_accuracy ~= 0.5807` in log; exact metadata `0.5806631445884705`)
- best observed validation loss in the rounded log is at Epoch 76 (`1.2239`); 15 subsequent epochs without a strict improvement terminate at Epoch 91, consistent with the registered `val_loss`, patience-15 lifecycle
- TensorFlow reports one internally compiled XLA cluster at run start; this is allowed by the preregistered runtime note boundary
- no traceback/exception/training error was found in the supplied complete training log

### Accepted endpoints

Selected checkpoint:

- clean-train accuracy: `0.614894284022432`
- clean-train macro-F1: `0.5614369765915708`
- clean-train loss: `1.1261594824768282`
- validation accuracy: `0.5806631373641683`
- validation macro-F1: `0.5238975323290902`
- validation loss: `1.2291889047018165`
- accuracy gap: `3.4231146658263656 pp`
- macro-F1 gap: `3.753944426248057 pp`

Independently recomputed deltas vs locked v1.1:

- validation accuracy: `+0.7244357759821685 pp`
- validation macro-F1: `+0.8472753182286907 pp`
- clean-train accuracy: `+0.9509213138736983 pp`
- clean-train macro-F1: `+1.3962141877976397 pp`

Applying the frozen precedence independently gives:

`TOKENIZER_UNDERFIT_REMAINS`

No PASS/STRETCH, OVERFIT_SHIFT, STRONG_SIGNAL, or PARTIAL_SIGNAL criterion is met; both clean-train accuracy `<0.70` and validation accuracy `<0.64`, so the registered underfit label applies.

### Scientific interpretation

The local-tokenizer-depth hypothesis receives only weak practical support: the extra shared `Dense(96->96)` improves both fitting and validation slightly, but by less than 1 pp accuracy and leaves the model strongly below the registered capacity/validation thresholds. The small clean train/validation gaps remain a positive generalization property, but they are not evidence that the architecture is sufficiently expressive.

In contrast, v1.0 -> v1.1 resolution produced the materially larger movement. The current evidence therefore does **not** justify another tokenizer-depth sweep. The next experiment should move to a different single bottleneck while preserving the successful high-resolution generic patches.

Non-blocking metadata note: `resolved_config.json` retains the implementation-status string from the reviewed candidate. This does not alter the scientific result because the exact reviewed source was preregistered and the independent wrapper/result objects explicitly record a complete valid scientific execution.

### Review decision

`CF_HPG_V1_2_TOKENIZER_SEED42_ACCEPTED`

Issue #50 is scientifically complete. No retry is authorized. Test remains untouched.
